app/src/main/java/mozilla/lockbox/store/DataStore.kt

```diff
@@ -93,8 +99,36 @@ open class DataStore(
                     is DataStoreAction.Reset -> reset()
                     is DataStoreAction.UpdateCredentials -> updateCredentials(action.syncCredentials)
                 }
+
+            }
+            .addTo(compositeDisposable)
+
+        setupAutoLock()
+    }
+
+    private fun setupAutoLock() {
+        lifecycleStore.lifecycleEvents
+            .filter { it == LifecycleAction.Background && stateSubject.value == State.Unlocked }
+            .subscribe {
+                autoLockSupport.storeNextAutoLockTime()
             }
             .addTo(compositeDisposable)
+
+        // backdate the lock timer when receiving manual lock actions
```

Comment:
I love the separation of concerns in this method. It makes it much easier to read and understand the explicit states/actions that we care about 💯